demo/list-box-presentation-demos.html

```diff
@@ -0,0 +1,90 @@
+<dom-module id="list-box-presentation-demos">
+  <template>
+    <style include="vaadin-component-demo-shared-styles">
+      :host {
+        display: block;
+      }
+    </style>
+
+    <h3>Separators</h3>
+    <vaadin-demo-snippet id="list-box-presentation-demos-separators">
+      <template preserve-content>
+        <vaadin-list-box>
+          <vaadin-item disabled>Weekdays</vaadin-item>
+          <hr>
+          <vaadin-item>Monday</vaadin-item>
+          <vaadin-item>Tuesday</vaadin-item>
+          <vaadin-item>Wednesday</vaadin-item>
+          <vaadin-item>Thursday</vaadin-item>
+          <vaadin-item>Friday</vaadin-item>
+          <hr>
+          <vaadin-item>Saturday</vaadin-item>
+          <vaadin-item>Sunday</vaadin-item>
+        </vaadin-list-box>
+      </template>
+    </vaadin-demo-snippet>
+
+    <h3>Customizing listbox item label</h3>
+    <vaadin-demo-snippet id="list-box-presentation-demos-custom-label">
+      <template preserve-content>
+        <b>User</b>
```

Comment:
Have we used `<b>` as a similar label in other demos? Imo this should be changed e.g. to `<label>User</label>` instead so it has some semantic meaning if we want to have a "label" here. Or we could just remove it. Using `<label>` it can also be styled to be bold and look the same by adding a style module in the `index.html`.

Reply:
Done.

Reply:
"The <label> tag defines a label for a <button>, <input>, <meter>, <output>, <progress>, <select>, or <textarea> element." Wouldn't it be out of context here?

Reply:
@tomivirkki Looking at the HTML living standard https://html.spec.whatwg.org/multipage/forms.html#the-label-element  
or the relevant MDN page: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

It is pretty clear that a `<label>` "can be" associated with a "labelable element" (those which you listed, plus form-associated custom elements) but doesn't have to be.

> The `label` element represents a caption in a user interface. The caption can be associated with a specific form control, known as the `label` element's **labeled control**, either using the `for` attribute, or by putting the form control inside the `label` element itself.
>
> Except where otherwise specified by the following rules, a label element has no labeled control.

So I think it's perfectly fine in this kind of context. Another alternative would be one of `h1`-`h6` but in this case I think `<label>` is more semantically correct since it "represents a caption in a user interface" and not a heading for a section.

Reply:
See the discussion at https://stackoverflow.com/questions/12195052/is-it-allowed-to-use-label-tag-without-labeled-control

Reply:
This discussion was continued at new PR https://github.com/vaadin/vaadin-list-box/pull/63#pullrequestreview-311572853
Marking this older "thread" as resolved.